### ISSUE TITLE
[feature request] Multi-Triggers for Contextual Habit Notifications (WiFi, Geolocation, and More)

### ISSUE THREAD

**Issue by adan-ea** [FriesI23/mhabit#71]
Hi, I would love to be able to receive a notification as soon as I connect to my home WiFi instead of a fixed hour

**Comment by FriesI23** [comment]
Because the appearance of IN-APP notifications depends on whether this app was running in background. It's actually related to Android's app hibernation mechanism. 

Perhaps you could try allow this app to stay in the background or just enable automatic startup for this app.

**Comment by adan-ea** [comment]
I meant : 
you have a task, let's say "Lock front door" 
Instead of having it set everyday at 5pm, you set it to trigger when you connect back to your home wifi, cause you could get home at 5:30 or later and the notification would be pointless
And to avoid spam it would be nice to chose "one time trigger" or a time frame 
So whenever i get home, my phone connect to wifi and i get the notification "lock door"

**Comment by Miaaaim** [comment]
I agree that reminding yourself to lock the door as soon as you get back home can help establish the habit of locking the door.  However, there are two reasons for not achieving this feature in the short term:

1. The ultimate goal of this App is to make your habits an automatic behavior without relying on external reminders. For example, over time, you'll find that locking the door becomes a natural part of your getting back home routine, requiring less conscious attention.
2. This feature is not simple. If you want the reminders to be in the right place, for example, you should set a time range for getting back home and the Wi-Fi's name. Otherwise, it may remind you at the wrong time or in the wrong place.
